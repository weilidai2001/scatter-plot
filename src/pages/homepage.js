import React from 'react';
import { ScatterplotChart } from 'react-easy-chart';

const data = [
    {
        type: 'One',
        x: 1,
        y: 5
    },
    {
        type: 'Two',
        x: 3,
        y: 1
    },
    {
        type: 'Three',
        x: 0,
        y: 6
    },
    {
        type: 'Four',
        x: 5,
        y: 2
    },
    {
        type: 'Five',
        x: 4,
        y: 4
    },
    {
        type: 'Six',
        x: 5,
        y: 9
    },
    {
        type: 'Seven',
        x: 9,
        y: 1
    },
    {
        type: 'Eight',
        x: 5,
        y: 6
    },
    {
        type: 'Nine',
        x: 3,
        y: 9
    },
    {
        type: 'Ten',
        x: 7,
        y: 9
    }
];

export default class Homepage extends React.Component {
    render() {
        return (
            <ScatterplotChart data={data} width={160} height={90} />
        )
    }
}