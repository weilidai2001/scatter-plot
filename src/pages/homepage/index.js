import React, { PureComponent, PropTypes } from 'react';
import { connect } from 'react-redux';
import { ScatterplotChart } from 'react-easy-chart';
import Loader from 'react-loader';

class Homepage extends PureComponent {
    static propTypes = {
        minX: PropTypes.number,
        maxX: PropTypes.number,
        minY: PropTypes.number,
        maxY: PropTypes.number,
        points: PropTypes.array,
        ajaxStatus: PropTypes.bool
    }

    render() {
        const { minX, maxX, minY, maxY, points, ajaxStatus } = this.props;

        return (
            <Loader loaded={!ajaxStatus}>
                <ScatterplotChart
                    data={points}
                    axes
                    axisLabels={{ x: 'X', y: 'Y' }}
                    width={document.body.clientWidth - 100}
                    height={500}
                    xDomainRange={[minX, maxX]}
                    yDomainRange={[minY, maxY]}
                />
            </Loader>
        );
    }
}

function mapStateToProps(state) {
    return {
        ajaxStatus: state.ajaxStatus,
        ...state.chartData
    };
}

export default connect(mapStateToProps)(Homepage);
