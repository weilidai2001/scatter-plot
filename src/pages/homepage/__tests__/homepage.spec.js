import React from 'react';
import { shallow } from 'enzyme';
import homepage from '../';

const PROPS = {
    minX: 0,
    maxX: 100,
    minY: 0,
    maxY: 100,
    points: [],
    ajaxStatus: false
};

describe('homepage UI', () => {
    it('should render', () => {
        const wrapper = shallow(<homepage {...PROPS} />);
        expect(wrapper).toMatchSnapshot();
    });
});
