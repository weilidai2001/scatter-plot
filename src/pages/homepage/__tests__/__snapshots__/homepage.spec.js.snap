// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage UI should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <homepage
    ajaxStatus={false}
    maxX={100}
    maxY={100}
    minX={0}
    minY={0}
    points={Array []}
/>,
  "nodes": Array [
    <homepage
      ajaxStatus={false}
      maxX={100}
      maxY={100}
      minX={0}
      minY={0}
      points={Array []}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <homepage
    ajaxStatus={false}
    maxX={100}
    maxY={100}
    minX={0}
    minY={0}
    points={Array []}
/>,
}
`;
